### PR TITLE
BIGTOP-3728: Docker dev/test env for centos7

### PR DESCRIPTION
**Pull request by kevinw66**
### Description of PR
Build `ambari-server` `ambari-agent-01` `ambari-agent-02` in docker, for easy develop and test

### NOTICE
Only test in centos7

### USAGE
* First time use this, or every time you update Docker file, run `build-image.sh`
* run `build-containers.sh` to build `ambari-server` `ambari-agent-01` `ambari-agent-02` container with network `ambari`
* every time you update scripts under `/services`, run `sync-scripts.sh` to copy your scripts to `ambari-server` `ambari-agent-01` `ambari-agent-02`
* after your dev/test finished, run `clear-containers.sh` to clear all related resources

3306 5005 8080 is exposed to your local machine.
you can visit localhost:8080 to access ambari-admin page.
you can connect localhost:3306 to see the data create by ambari.
you can connect localhost:5005 via Intellij IDEA to debug ambari-server.

### TEST RESULTS FOR INSTALL COMPONENTS
Remember to add `ambari-server` `ambari-agent-01` `ambari-agent-02` to your hosts
bigtop-image-1

Components successfully installed.
bigtop-image-2
bigtop-image-6

### TEST RESULTS FOR ENABLE KERBEROS
Admin password is `admin`
bigtop-image-7

Remember to comment this line for high version of JDK 1.8
bigtop-image-4

Kerberos successfully enabled.
bigtop-image-5
bigtop-image-8
bigtop-image-9

### CODE REVIEW

**Comment by vivostar** [comment]
+1, It work well on my centos7 machine.
It's a useful dev tools for mpack developing, thanks @kevinw66 
image

File: bigtop-packages/src/common/bigtop-ambari-mpack/dev-support/docker/centos7/build-docker.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+docker exec ambari-server bash -c "mysql -e \"GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' IDENTIFIED BY 'root' WITH GRANT OPTION\""
+docker exec ambari-server bash -c "mysql -e \"DROP USER ''@'localhost'\""
+docker exec ambari-server bash -c "mysql -e \"DROP USER ''@'ambari-server'\""
+docker exec ambari-server bash -c "mysql -e \"DROP DATABASE test\""
+docker exec ambari-server bash -c "mysql -e \"CREATE DATABASE ambari\""
+docker exec ambari-server bash -c "mysql --database=ambari -e  \"source /var/lib/ambari-server/resources/Ambari-DDL-MySQL-CREATE.sql\""
+docker exec ambari-server bash -c "mysql -e \"FLUSH PRIVILEGES\""
+
+echo -e "\033[32mSetting up ambari-server\033[0m"
+docker exec ambari-server bash -c "ambari-server setup --java-home=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.332.b09-1.el7_9.x86_64 --database=mysql --databasehost=localhost --databaseport=3306 --databasename=ambari --databaseusername=root --databasepassword=root -s"
```
Comment (guyuqi) [inline bigtop-packages/src/common/bigtop-ambari-mpack/dev-support/docker/centos7/build-docker.sh:24]: Mpack should support multi-arch.
Please replace `x86_64` with `$basearch` in `java-1.8.0-openjdk-1.8.0.332.b09-1.el7_9.x86_64`.
`$basearch` could be obtained from environment variables.

**Comment by guyuqi** [comment]
It's pretty convenient for us to test Mapck.
Thanks for working on it.  @kevinw66 

Verified it on x86.
But there is something tricky on Arm64 that the `Maven` was not preinstalled in the aarch64 base docker image( `Centos:7`) .

So the issues occurred:

```
Installing Bigtop Ambari Mpack
./build-containers.sh: line 66: mvn: command not found
lstat /home/linux/yuqi/bigtop/bigtop-packages/src/common/bigtop-ambari-mpack/bgtp-ambari-mpack/target: no such file or directory
Using python  /usr/bin/python
Installing management pack
/usr/lib/ambari-server/lib/ambari_commons/subprocess32.py:156: RuntimeWarning: The _posixsubprocess module is not being used. Could not import it from ambari_commons.libs.aarch64 Child process reliability may suffer if your program uses threads.
  "program uses threads.", RuntimeWarning)
ERROR: Download /bgtp-ambari-mpack-1.0.0.0-SNAPSHOT-bgtp-ambari-mpack.tar.gz with python lib [urllib2] failed with error: (<type 'exceptions.ValueError'>, ValueError('unknown url type: /bgtp-ambari-mpack-1.0.0.0-SNAPSHOT-bgtp-ambari-mpack.tar.gz',), <traceback object at 0xffff9247cb00>)
Trying to download /bgtp-ambari-mpack-1.0.0.0-SNAPSHOT-bgtp-ambari-mpack.tar.gz to /var/lib/ambari-server/data/tmp/bgtp-ambari-mpack-1.0.0.0-SNAPSHOT-bgtp-ambari-mpack.tar.gz with [curl] command.
ERROR: Download file /bgtp-ambari-mpack-1.0.0.0-SNAPSHOT-bgtp-ambari-mpack.tar.gz with [curl] command failed with error: curl: (3) <url> malformed

ERROR: Unable to download file /bgtp-ambari-mpack-1.0.0.0-SNAPSHOT-bgtp-ambari-mpack.tar.gz!
ERROR: unable to donwload file /bgtp-ambari-mpack-1.0.0.0-SNAPSHOT-bgtp-ambari-mpack.tar.gz!
ERROR: Management pack could not be downloaded!
ERROR: Exiting with exit code -1.
REASON: Management pack could not be downloaded!
Using python  /usr/bin/python
Restarting ambari-server

```

File: bigtop-packages/src/common/bigtop-ambari-mpack/dev-support/docker/centos7/build-containers.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+docker exec ambari-server bash -c "systemctl enable krb5kdc"
+docker exec ambari-server bash -c "systemctl start kadmin"
+docker exec ambari-server bash -c "systemctl enable kadmin"
+# KDC HOST: ambari-server
+# REALM NAME: EXAMPLE.COM
+# ADMIN PRINCIPAL: admin/admin@EXAMPLE.COM
+# ADMIN PASSWORD: admin
+
+echo -e "\033[32mInstalling Bigtop Ambari Mpack\033[0m"
+mvn clean install -DskipTests -Drat.skip -f ../../../bgtp-ambari-mpack/pom.xml
```
Comment (guyuqi) [inline bigtop-packages/src/common/bigtop-ambari-mpack/dev-support/docker/centos7/build-containers.sh:66]: IMO, we should not build Mpack on local host directly. 
The build may affected by potential dependency issues on different host Distros .( In my local ubuntu20.4 Arm64 host, maven package was not pre-installed).
Let's build Mpack by docker based on `bigtop/slaves` docker images.
`build-containers.sh
`:
```
#!/bin/bash
ARCH=`uname -m`
basearch=
if [ ${ARCH} == "aarch64" ]; then
    basearch="-aarch64"
elif [ ${ARCH} == "ppc64le" ]; then
    basearch="-ppc64le"
fi

.....
...
..
.
cd ../../../../../../../
docker run --rm -v `pwd`:/ws --workdir /ws bigtop/slaves:trunk-centos-7${basearch} bash -c '. /etc/profile.d/bigtop.sh; ./gradlew bigtop-utils-clean bigtop-ambari-mpack-clean bigtop-utils-pkg bigtop-ambari-mpack-pkg'
docker cp output/bigtop-ambari-mpack/noarch/bigtop-ambari-mpack-2.7.5.0-*.rpm ambari-server:/
docker cp output/bigtop-utils/noarch/bigtop-utils-*.rpm ambari-server:/
cd -

docker exec ambari-server bash -c "rpm -ivh -f /bigtop-utils-*.rpm; rpm -ivh -f /bigtop-ambari-mpack-2.7.5.0-1.el7.noarch.rpm"
docker exec ambari-server bash -c "cp /usr/lib/bigtop-ambari-mpack/bgtp-ambari-mpack-1.0.0.0-SNAPSHOT-bgtp-ambari-mpack.tar.gz /"

# Then run:  "ambari-server install-mpack" ......

```
We build Mapck rpm by docker.  And push rpm into containers and install mpack rpm.
Installation of Mpack rpm is required by bigtop-utils-pkg.  So we also build bigtop-utils-pkg here.
Alternatively, we could push bigtop-utils-pkg installation into Dockfile.

Comment (kevinw66) [reply bigtop-packages/src/common/bigtop-ambari-mpack/dev-support/docker/centos7/build-containers.sh]: Thanks, I have updated the scripts.

**Review by guyuqi — APPROVED** [review]
Tested the PR both for x86 and Arm64 on Cetnos-7.  
+1, LGTM now.  
It's found that the licensed info was also added in the latest commit which could be avoid `RAT` issues.
Thanks, @kevinw66

**Pull request merged**